client.go

```diff
@@ -1,40 +1,126 @@
 package ntrip
 
 import (
+	"fmt"
 	"io"
-	"net"
+	"io/ioutil"
 	"net/http"
-	"strings"
+	"time"
 )
 
-// TODO: Consider whether it would be better for the Client / Server API to hang of a type much like
-//  the paho mqtt clients
+type Client interface {
+	GetSourcetable() (Sourcetable, error)
+	GetMount(mount string) (io.ReadCloser, error)
+	// TODO: PostMount as part of ntrip.Client doesn't really fit the NTRIP naming, but would likely be a better interface for this library
+}
 
-// NewClientRequestV2 constructs an http.Request which can be used as an NTRIP v2 Client
-func NewClientRequestV2(url string) (*http.Request, error) {
-	req, err := http.NewRequest(http.MethodGet, url, nil)
-	req.Header.Set("User-Agent", "NTRIP go-gnss/ntrip/client")
+type ClientV2 struct {
+	http.Client
+	url      string
+	Username string
+	Password string
```

Comment:
ClientV2 will likely end up with other ways of authorizing requests, but I'm not yet sure how to abstract that.

Reply:
The http library gives you SetBasicAuth for requests, but to use other auth methods you have to set the authorization header yourself. Perhaps we can do the same thing. Allow the client to pass in different request headers per request.

Reply:
Noting there that the http library has a Request object which you can interact with, where we don't currently - so would have to be an argument to the GetMount method (GetSourcetable too I guess).